ETC Course: 4. Satoshi Nakamoto and Bitcoin (#1052)
ethereumclassic/ethereumclassic.github.io: content/blog/2023-01-12-ethereum-classic-course-4-satoshi-nakamoto-and-bitcoin/index.md

**4. Protection of all the history of transactions:** For any transaction that has been sent and included in a block in Bitcoin to be reversed or deleted the same amount of work that was done for its inclusion must be done again. This prevents any attacker from reversing past transactions, and the older a transaction is the more difficult it is reverse or delete.

## Nakamoto Consensus

The system invented by Satoshi Nakamoto is called Nakamoto Consensus and includes all the components described above:

1. Full transmission 
2. Full replication
3. Proof of work
4. Block production
5. Fixed supply

And, all these combined assure the immutability and security of the blockchain.

## What Does it Accomplish?

As said before, Nakamoto Consensus accomplishes a level of safety and security of the system never reached before while keeping it decentralized.

This makes it trust minimized, meaning that users may interact directly between themselves in a peer-to-peer way, sending and receiving money, without the need of trusted third parties, who may be captured by special interests, restricting freedom and human action.

> Just as the internet, Bitcoin is designed to survive a nuclear war, and, on top of that, it assures hard money for the internet.

Reviewer: Line 99: Corrected run-on sentence
```suggestion
Just like the internet, Bitcoin is designed to survive a nuclear war. On top of that, it assures hard money for the internet.
```

## Implications for Ethereum Classic

Ethereum Classic uses the same consensus mechanism as Bitcoin. However, ETC adds to the base technology of Bitcoin the ability to store and execute smart contracts or decentralized programs. This turns applications in ETC into decentralized applications which execute their code inside the highly secure environment of the blockchain. 

> So, ETC has the same two base pillars as Bitcoin; a fixed supply and the proof of work based consensus mechanism; but to that it added smart contracts. This makes ETC’s principle of Code Is Law a true statement. Therefore, ETC gives money and applications the benefits of soundness, survivability, censorship resistance, unstoppability, and they are uncompromisable and uncapturable.

Reviewer: Line 105: Corrected run-on sentence
```suggestion
So, ETC has the same two base pillars as Bitcoin; a fixed supply and the proof of work based consensus mechanism; but to that it added smart contracts. This makes ETC’s principle of Code Is Law a true statement. Therefore, ETC gives money and applications the benefits of soundness, survivability, censorship resistance, and unstoppability making them uncompromisable and uncapturable.
```
